src/graphics/grab-pass.js

```diff
@@ -0,0 +1,158 @@
+import { ADDRESS_CLAMP_TO_EDGE, FILTER_LINEAR, FILTER_LINEAR_MIPMAP_LINEAR, PIXELFORMAT_R8_G8_B8, PIXELFORMAT_R8_G8_B8_A8 } from "./constants.js";
+import { RenderTarget } from "./render-target.js";
+import { Texture } from "./texture.js";
+
+// A class used by WebGl device to handle the capture of the current framebuffer into a texture
+// to be used by following draw calls to implement refraction
+class GrabPass {
+    constructor(device, useAlpha) {
+        this.device = device;
+        this.useAlpha = useAlpha;
+
+        // on webgl1 we cannot mipmap not power of two texture, so use additional power of two texture so disable it for now
```

Comment:
This comment doesn't make sense when I read it back. Can you revise it please?

Reply:
done, thanks.